tests/ui/needless_move.rs

```diff
@@ -0,0 +1,1896 @@
+            let mut cx = Context::from_waker(&waker);
+
+            // Run the future to completion.
+            loop {
+                match fut.as_mut().poll(&mut cx) {
+                    Poll::Ready(res) => return res,
+                    Poll::Pending => thread::park(),
+                }
+            }
+        }
+
+        async fn bar() {
+            panic!()
+        }
+
+        async fn foo() {
+            bar().await
+        }
+
+        #[track_caller]
+        async fn bar_track_caller() {
+            panic!()
+        }
+
+        async fn foo_track_caller() {
+            bar_track_caller().await
+        }
+
+        struct Foo;
+
+        impl Foo {
+            #[track_caller]
+            async fn bar_assoc() {
+                panic!();
+            }
+        }
+
+        async fn foo_assoc() {
+            Foo::bar_assoc().await
+        }
+
+        // Since compilation is expected to fail for this fn when using
+        // `nofeat`, we test that separately in `async-closure-gate.rs`
+        #[cfg(cls)]
+        async fn foo_closure() {
+            let c = #[track_caller]
+            async || {
+                panic!();
+            };
+            c().await
+        }
+
+        // Since compilation is expected to fail for this fn when using
+        // `nofeat`, we test that separately in `async-block.rs`
+        #[cfg(cls)]
+        async fn foo_block() {
+            let a = #[track_caller]
+            async {
+                panic!();
+            };
+            a.await
+        }
+
+        fn panicked_at(f: impl FnOnce() + panic::UnwindSafe) -> u32 {
+            let loc = Arc::new(Mutex::new(None));
+
+            let hook = panic::take_hook();
+            {
+                let loc = loc.clone();
+                panic::set_hook(Box::new(move |info| {
+                    *loc.lock().unwrap() = info.location().map(|loc| loc.line())
+                }));
+            }
+            panic::catch_unwind(f).unwrap_err();
+            panic::set_hook(hook);
+            let x = loc.lock().unwrap().unwrap();
+            x
+        }
+
+        fn main() {
+            assert_eq!(panicked_at(|| block_on(foo())), 46);
+
+            #[cfg(afn)]
+            assert_eq!(panicked_at(|| block_on(foo_track_caller())), 61);
+            #[cfg(any(cls, nofeat))]
+            assert_eq!(panicked_at(|| block_on(foo_track_caller())), 57);
+
+            #[cfg(afn)]
+            assert_eq!(panicked_at(|| block_on(foo_assoc())), 76);
+            #[cfg(any(cls, nofeat))]
+            assert_eq!(panicked_at(|| block_on(foo_assoc())), 71);
+
+            #[cfg(cls)]
+            assert_eq!(panicked_at(|| block_on(foo_closure())), 84);
+
+            #[cfg(cls)]
+            assert_eq!(panicked_at(|| block_on(foo_block())), 96);
+        }
+    }
+
+    // rust/tests/ui/async-await/deep-futures-are-freeze.rs
+
+    fn _deep_futures_are_freeze() {
+        // no-prefer-dynamic
```

Comment:
I would like to ensure `x test` won't read these, somehow, I dunno, very cautious 😅 I doubt it does when running tests in subtrees or when they aren't at the top of the file

Or just remove them